spec/namespaces/query_group.yaml

```diff
@@ -0,0 +1,135 @@
+      x-version-added: '2.16'
+      description: Deletes the specified QueryGroups or delete all if no name is provided.
+      responses:
+        '200':
+          $ref: '#/components/responses/query_group.delete@200'
+  /_query_group/{name}:
+    get:
+      operationId: query_group.get.1
+      x-operation-group: query_group.get
+      x-version-added: '2.16'
+      description: Gets the specified QueryGroup or get all if no name is provided.
+      parameters:
+        - $ref: '#/components/parameters/query_group.get::path.name'
+      responses:
+        '200':
+          $ref: '#/components/responses/query_group.get@200'
+    put:
+      operationId: query_group.put.1
+      x-operation-group: query_group.put
+      x-version-added: '2.16'
+      description: Creates or updates a specified query group.
+      parameters:
+        - $ref: '#/components/parameters/query_group.put::path.name'
+      requestBody:
+        $ref: '#/components/requestBodies/query_group.put.1'
+      responses:
+        '200':
+          $ref: '#/components/responses/query_group.put@200'
+    delete:
+      operationId: query_group.delete.1
+      x-operation-group: query_group.delete
+      x-version-added: '2.16'
+      description: Deletes the specified QueryGroups or delete all if no name is provided.
+      parameters:
+        - $ref: '#/components/parameters/query_group.delete::path.name'
+      responses:
+        '200':
+          $ref: '#/components/responses/query_group.delete@200'
+components:
+  requestBodies:
+    query_group.put.0:
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              $ref: '../schemas/query_group._common.yaml#/components/schemas/QueryGroupCreateStructure'
+      required: true
+    query_group.put.1:
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              $ref: '../schemas/query_group._common.yaml#/components/schemas/QueryGroupUpdateStructure'
+      required: true
+  responses:
+    query_group.put@200:
+      description: ''
+      content:
+        application/json:
+          schema:
+            $ref: '../schemas/query_group._common.yaml#/components/schemas/QueryGroupResponseStructure'
+    query_group.delete@200:
+      description: ''
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              deleted:
+                type: array
+                items:
+                  $ref: '../schemas/query_group._common.yaml#/components/schemas/QueryGroupResponseStructure'
+    query_group.get@200:
+      description: ''
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              query_groups:
+                type: array
+                items:
+                  $ref: '../schemas/query_group._common.yaml#/components/schemas/QueryGroupResponseStructure'
```

Comment:
Is the word Structure coming from Smithy? It doesn't add any info and should be removed. These names will be generated in some clients. So we're want to keep these short and simple.